src/NzbDrone.Core/Datastore/Migration/008_redacted_api.cs

```diff
@@ -0,0 +1,75 @@
+using System.Data;
+using FluentMigrator;
+using Newtonsoft.Json.Linq;
+using NzbDrone.Common.Serializer;
+using NzbDrone.Core.Datastore.Migration.Framework;
+
+namespace NzbDrone.Core.Datastore.Migration
+{
+    [Migration(8)]
+    public class redacted_api : NzbDroneMigrationBase
+    {
+        protected override void MainDbUpgrade()
+        {
+            Execute.WithConnection(MigrateToRedactedApi);
+        }
+
+        private void MigrateToRedactedApi(IDbConnection conn, IDbTransaction tran)
+        {
+            using (var cmd = conn.CreateCommand())
+            {
+                cmd.Transaction = tran;
+                cmd.CommandText = "SELECT Id, Settings FROM Indexers WHERE Implementation = 'Redacted'";
+
+                using (var reader = cmd.ExecuteReader())
+                {
+                    while (reader.Read())
+                    {
+                        var id = reader.GetInt32(0);
+                        var settings = reader.GetString(1);
+                        if (!string.IsNullOrWhiteSpace(settings))
+                        {
+                            var jsonObject = Json.Deserialize<JObject>(settings);
+
+                            // Remove username
+                            if (jsonObject.ContainsKey("username"))
+                            {
+                                jsonObject.Remove("username");
+                            }
+
+                            // Remove password
+                            if (jsonObject.ContainsKey("password"))
+                            {
+                                jsonObject.Remove("password");
+                            }
+
+                            // Add apikey
+                            if (!jsonObject.ContainsKey("apikey"))
```

Comment:
Remove this and the add for passkey below. They are not needed and cause the first time run after migration to skip validation if no key is present

Reply:
Good call, I removed both blocks.